scripts/bash-completion/bpftrace

```diff
@@ -0,0 +1,90 @@
+# bash completion of bpftrace
+
+_bpftrace_filedir()
+{
+  # bash-completion 2.11-1091-g6e8a1546bde2 rename _filedir to _comp_compgen_filedir
+  if [[ ${BASH_COMPLETION_VERSINFO[0]} == 2 ]] && \
+     [[ ${BASH_COMPLETION_VERSINFO[1]} > 11 ]]; then
+    _comp_compgen_filedir "${@}"
+  else
+    _filedir "${@}"
+  fi
+}
+
+_bpftrace()
+{
+  local cur prev words
+
+  _init_completion -- "$@" || return
+
+  local all_args='-B -f -o -e -h --help -I --include -l -p -c
+                  --usdt-file-activation --unsafe -q --info -k -kk
+                  -V --version --no-warnings -v --dry-run -d
+                  --emit-elf --emit-llvm'
+
+  case ${prev} in
+  -B)
+    COMPREPLY=( $(compgen -W "line full none" -- ${cur}) )
+    return 0
+    ;;
+  -f)
+    COMPREPLY=( $(compgen -W "text json" -- ${cur}) )
+    return 0
+    ;;
+  -c)
+    COMPREPLY=( $(compgen -c -- ${cur}) )
+    return 0
+    ;;
+  -o | -I | --include | --emit-elf | --emit-llvm)
+    _bpftrace_filedir
+    return 0
+    ;;
+  -I)
+    _bpftrace_filedir -d
+    return 0
+    ;;
+  -e)
+    # TODO: or replace '${PROBES} ${ITERS}' to '$(sudo bpftrace -l)', however,
+    # `sudo' maybe need password, and 'bpftrace -l' is too slow, could be a
+    # better way?
+    local PROBES="BEGIN END kprobe kretprobe tracepoint fentry fexit uprobe uretprobe usdt"
+    local ITERS="iter:bpf_link
+                 iter:bpf_map
+                 iter:bpf_map_elem
+                 iter:bpf_prog
+                 iter:bpf_sk_storage_map
+                 iter:cgroup
+                 iter:ipv6_route
+                 iter:ksym
+                 iter:netlink
+                 iter:sockmap
+                 iter:task
+                 iter:task_file
+                 iter:task_vma
+                 iter:tcp
+                 iter:udp
+                 iter:unix"
+    COMPREPLY=( $(compgen -W "${PROBES} ${ITERS}" -- ${cur}) )
+    return 0
```

Comment:
This is very interesting. Few thoughts:

1. Completing `-e` seems quite tricky. But maybe adds a lot of value for 1-liner use case, where there's only a single attachpoint. I wonder if we could do some caching to speed things up? And if bpftrace is slow at this, maybe profile the listing codepath and see if there's any easy wins to be had.
2. Completing `-l` could be quite useful as well. Probably no extra code if we're already doing `-e`
3. Hard coding stuff here does not seem like a good idea. Could confuse users, especially if we complete stuff that is not available on running kernel.

Reply:
I just fix other problems, Maybe we can use this method first and modify it later when we have a better solution, is that ok?

Reply:
And, i just add `MISCS` for complete `-c`. By the way, like `bash -c "ls -l"`, bash-completion do nothing for `bash -c`, or we just ignore `-c` here?